src/Microsoft.AspNetCore.Mvc.Core/Routing/UrlHelperCommon.cs

```diff
@@ -0,0 +1,261 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+
+namespace Microsoft.AspNetCore.Mvc.Routing
+{
+    internal class UrlHelperCommon
```

Comment:
Consider calling this UrlHelperBase and just making it an abstract base class

Reply:
Right now it is more of a helper class to be used internally by the two implementations, rather than a base class.

Would it be a breaking change to have it as a base class that the existing public `UrlHelper` inherits from?

Reply:
> Would it be a breaking change to have it as a base class that the existing public UrlHelper inherits from?

Since the base would be implementing IUrlHelper, I think we are good.